rust/hyperlane-base/src/server/eigen_node.rs

```diff
@@ -0,0 +1,303 @@
+            .set(42);
+
+        let node_api = EigenNodeAPI::new(core_metrics.clone());
+        let app = node_api.router();
+
+        // Run the app in the background using a test server
+        let server =
+            axum::Server::bind(&"127.0.0.1:0".parse().unwrap()).serve(app.into_make_service());
+        let addr = server.local_addr();
+        let server_handle = tokio::spawn(server);
+
+        // Create a client and make a request to the `/node/health` endpoint
+        // if signed_checkpoint - observed_checkpoint > 10 return 503 - unhealthy
+        let client = reqwest::Client::new();
+        let res = client
+            .get(format!("http://{}/node/health", addr))
+            .send()
+            .await
+            .expect("Failed to send request");
+        assert_eq!(res.status(), StatusCode::SERVICE_UNAVAILABLE);
+
+        // if signed_checkpoint - observed_checkpoint <= 10 return 206 - partially healthy
+        core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_processed", "ethereum"])
+            .set(34);
+        let res = client
+            .get(format!("http://{}/node/health", addr))
+            .send()
+            .await
+            .expect("Failed to send request");
+        assert_eq!(res.status(), StatusCode::PARTIAL_CONTENT);
+
+        // if signed_checkpoint - observed_checkpoint <= 1 return 200 - healthy
+        core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_processed", "ethereum"])
+            .set(42);
+        let res = client
+            .get(format!("http://{}/node/health", addr))
+            .send()
+            .await
+            .expect("Failed to send request");
+        assert_eq!(res.status(), StatusCode::OK);
+
+        // Stop the server
+        server_handle.abort();
+    }
+
+    #[tokio::test]
+    async fn test_eigen_node_services_handler() {
+        let core_metrics =
+            Arc::new(CoreMetrics::new("dummy_relayer", 37582, Registry::new()).unwrap());
+
+        let node_api = EigenNodeAPI::new(core_metrics);
+        let app = node_api.router();
+
+        // Run the app in the background using a test server
+        let server =
+            axum::Server::bind(&"127.0.0.1:0".parse().unwrap()).serve(app.into_make_service());
+        let addr = server.local_addr();
+        let server_handle = tokio::spawn(server);
+
+        // Create a client and make a request to the `/node/services` endpoint
+        let client = reqwest::Client::new();
+        let res = client
+            .get(format!("http://{}/node/services", addr))
+            .send()
+            .await
+            .expect("Failed to send request");
+
+        // Check that the response status is OK
+        assert_eq!(res.status(), StatusCode::OK);
+
+        // check the response body if needed
+        let json: Value = res.json().await.expect("Failed to parse json");
```

Comment:
ditto here, maybe would be easier to parse the JSON into the type instead of the handcrafted indexing

Reply:
fixed